bson/bsoncodec/time_codec.go

```diff
@@ -73,6 +65,18 @@ func (tc *TimeCodec) DecodeValue(dc DecodeContext, vr bsonrw.ValueReader, val re
 		if err != nil {
 			return err
 		}
+	case bsontype.Int64:
+		i64, err := vr.ReadInt64()
+		if err != nil {
+			return err
+		}
+		timeVal = time.Unix(i64/1000, i64%1000*1000000)
+	case bsontype.Timestamp:
+		t, _, err := vr.ReadTimestamp()
```

Comment:
@craiggwilson Can you verify that this timestamp logic is correct and we should be ignoring the second return value from `ReadTimestamp`?

Reply:
Why isn't this just an mgo compatibility question rather than a question for Craig, i.e.  what does mgo do?

Reply:
Decoding int64 and timestamp isn't an mgo compatibility change, mgo didn't decode these bsontypes into time.Time

Reply:
I'd be fine leaving them out until someone asks for it.  Regarding timestamp specifically, it's not widely used and bsonspec.org says "BSON has a special timestamp type for internal MongoDB use"

Reply:
I'd rather put them in before users ask for them, they're fairly low risk and don't take much work. Our internal users may want them, and Craig actually suggested them: https://github.com/mongodb/mongo-go-driver/pull/201#discussion_r339729595